test/fluree/db/policy/target_test.clj

```diff
@@ -484,3 +485,31 @@
                                      "select"   {"?s" ["*"]}
                                      "opts"     {"policy" [default-policy classification-policy]}}))
               "only data with classification < 1 should be visible when using opts.policy"))))))
+
+(deftest ^:integration target-policy-exception
+  (testing "Invalid policy target queries throw exceptions"
+    (let [conn   @(fluree/connect-memory)
+          ledger @(fluree/create conn "policy/target-policy-exception")
+          db0    (fluree/db ledger)
+          db1    @(fluree/stage db0 {"@context" {"a" "http://a.co/"}
+                                     "insert"
+                                     [{"@id"    (:id arnold)
+                                       "a:name" "Arnold"}
+                                      {"@id"    (:id burt)
+                                       "a:name" "Burt"}
+                                      {"@id"    (:id charles)
+                                       "a:name" "Chuck"}]})
+          bad-policy [{"@context" {"ex" "http://example.org/ns/"
+                                   "f"  "https://ns.flur.ee/ledger#"}
+                       "@id"             "ex:badSubjectPolicy"
+                       "@type"           ["f:AccessPolicy"]
+                       "f:targetSubject" {"@type"  "@json"
+                                          "@value" {"blah" {[] []}}}
+                       "f:action"        [{"@id" "f:view"}, {"@id" "f:modify"}]
+                       "f:query"         {"@type"  "@json"
+                                          "@value" {"@context" {"ex" "http://example.org/ns/"}
+                                                    "where"    [{"@id" "?$this" "ex:classification" "?c"}
+                                                                ["filter", "(< ?c 1)"]]}}}]
+          bad-db @(fluree/wrap-policy db1 bad-policy)]
+
+      (is (util/exception? bad-db)))))
```

Comment:
What sort of error message will the user see? Since policies may be parsed long after they are transacted, the user who sees this error may not be able to do anything to fix it.

Reply:
These errors are reported immediately when calling wrap-policy.

Do you mean if a user created policy rules in a DB but never used, them - then decided to use them later? To capture those errors we'd have to scan for policy transactions as they happen and try to parse them then.

I am not sure that is a good idea, but certainly debatable. Outside of the scope here - whose purpose is to fix the issue that policy errors were not getting reported at all.